galleryModule.js

```diff
@@ -0,0 +1,26 @@
+var galleryModule = (function() {
+  function makeGallery(config, imageName, imageSrc) {
+    var gallery = document.getElementById(config.gallery.slice(1));
+    var image = createImage(config, imageName, imageSrc); //image = return loadedImage from function createImage()
+    var link = createLink(imageSrc); //link = return loadedLink from function createLink()
+    link.appendChild(image); //append image as the last child of an <a> tag
+    gallery.appendChild(link); //append <a> tag as the last child of the gallery
+  }
+  function createImage(config, imageName, imageSrc) {
+    var loadedImage = document.createElement("img"); //create new image object
+    loadedImage.setAttribute("height", config.thumbHeight); //add attributes to the image: size, title, source
+    loadedImage.setAttribute("width", config.thumbWidth);
+    loadedImage.setAttribute("title", imageName);
+    loadedImage.setAttribute("src", imageSrc);

```

Comment:
obrazki powinnas tworzyc na canvasie. inaczej wrzucasz tam obrazek pelnej wielkosci i przy 100 duzych obrazkach Twoja strona zacznie mulic. ustawianie height / width to tak naprawde polsrodek - bo bedziesz miala po prostu wieksza gestosc pikseli a takie same zajete miejsce w pamieci.